unistd/file.c

```diff
@@ -147,67 +150,55 @@ int mkfifo(const char *filename, mode_t mode)
 }
 
 
-int symlink(const char *path1, const char *path2)
+int symlink(const char *target, const char *linkpath)
 {
 	oid_t dir;
-	char *canonical1, *canonical2;
+	char *canonical_linkpath;
 	char *dir_name, *name;
 	msg_t msg;
 	int len1, len2;
 	int ret;
 
-	if (path1 == NULL || path2 == NULL)
+	if (target == NULL || linkpath == NULL)
 		return -EINVAL;
 
-	canonical1 = canonicalize_file_name(path2);
-	if (canonical1 == NULL) {
-		return -ENOMEM;
+	/* allow_missing_leaf = 1 -> we will be creating a file */
+	canonical_linkpath = resolve_path(linkpath, NULL, 1, 1);
+	if (canonical_linkpath == NULL) {
+		return -1; /* errno set by resolve_path */
 	}
```

Comment:
No need for brackets

Reply:
the current code is quite inconsequential, I went for "minimal change" - so if there was single-statement if with brackets I left it as-is.

Changed in next patchset (will resolve when pushed)